src/dbnode/persist/fs/seek_manager.go

```diff
@@ -378,20 +390,21 @@ func (m *seekerManager) newOpenSeeker(
 		m.filePathPrefix,
 		m.opts.DataReaderBufferSize(),
 		m.opts.InfoReaderBufferSize(),
-		m.opts.SeekReaderBufferSize(),
 		m.bytesPool,
 		true,
-		nil,
 		m.opts,
 	)
 	seeker := seekerIface.(*seeker)
 
 	// Set the unread buffer to reuse it amongst all seekers.
 	seeker.setUnreadBuffer(m.unreadBuf.value)
 
-	if err := seeker.Open(m.namespace, shard, blockStart); err != nil {
+	resources := m.getSeekerResources()
+	if err := seeker.Open(m.namespace, shard, blockStart, m.getSeekerResources()); err != nil {
+		m.putSeekerResources(resources)
 		return nil, err
 	}
+	m.putSeekerResources(resources)
```

Comment:
Are you calling `m.getSeekerResources()` twice here?  Seems like the first one doesn't get used.

Did you want something more like this?:
```go
	resources := m.getSeekerResources()
	err := seeker.Open(m.namespace, shard, blockStart, resources)
	m.putSeekerResources(resources)

	if err != nil {
		return nil, err
	}

```

Reply:
yeah that was a mistake thanks